src/DummyTest.java

```diff
@@ -1,9 +1,139 @@
+
+        // Assert
+        assertEquals(100, entity.getHealth(), "Wrong health");
+        assertEquals(50, entity.getAttack(), "Wrong attack");
+        assertEquals(40, entity.getDefense(), "Wrong defense");
+        assertEquals(2, entity.getCooldown(), "Wrong cooldown");
+        assertEquals("Bob", entity.getName(), "Wrong name");
+        assertEquals("Bob story", entity.getLore(), "Wrong lore");
+    }
+
+    @Test
+    @DisplayName("Test if health is 0")
+    void testHealthIsZero() {
+        // Arrange
+        Entity entity;
+
+        //Act
+        entity = new Entity(0, 50, 40, 2, "Bob", "Bob story");
+        
+        // Assert
+        assertEquals(0, entity.getHealth(), "Wrong health");
+    }
+
+
+    @Test
+    @DisplayName("Test if entity is dead")
+    void testHealthIsBigger() {
+        // Arrange
+        Entity entity;
+
+        //Act
+        entity = new Entity(0, 20, 60, 2, "Bob", "Bob story");
+        
+        // Assert
+        assertEquals(false, entity.isAlive(), "Wrong health");
+    }
+
+    @Test
+    @DisplayName("Test if entity is alive")
+    void testEntityAlive() {
+        // Arrange
+        Entity entity;
+
+        //Act
+        entity = new Entity(5, 50, 40, 2, "Bob", "Bob story");
+        
+        // Assert
+        assertEquals(true, entity.isAlive(), "Wrong health");
+    }
+
+    @Test
+    @DisplayName("Test if cooldown bigger than 0")
+    void testCooldownBiggerThanZero() {
+        // Arrange
+        Entity entity;
+
+        //Act
+        entity = new Entity(100, 50, 40, 2, "Bob", "Bob story");
+        
+        // Assert
+        assertEquals(entity.getCooldown() >= 0, true, "Wrong cooldown");
+    }
+
+   
+    @Test
+    @DisplayName("Test if attack bigger than 0")
+    void testAttackBiggerThanZero() {
+        // Arrange
+        Entity entity;
+
+        //Act
+        entity = new Entity(100, 50, 40, 2, "Bob", "Bob story");
+        
+        // Assert
+        assertEquals(entity.getAttack() >= 0, true, "Wrong attack");
+    }
+
+    @Test
+    @DisplayName("Test if defense bigger than 0")
+    void testDefenseBiggerThanZero() {
+        // Arrange
+        Entity entity;
+
+        //Act
+        entity = new Entity(100, 50, 40, 2, "Bob", "Bob story");
+        
+        // Assert
+        assertEquals(entity.getDefense() >= 0, true, "Wrong defense");
+    }
+
+    @Test
+    @DisplayName("Test if health is below 0")
+    void exceptionTestingHealth() {
+        // Arrange
+        Entity entity;
+
+        //Act
+
+        //This assetion will fail
+        try{
+        entity = new Entity(-1, 50, 40, 2, "Bob", "Bob story");
+        
+        // Exception exception = assertThrows(
+        //     IllegalArgumentException.class,
+        //     () -> entity.getHealth());
+        } catch (IllegalArgumentException e) {
+            // Assert
+            assertEquals("Health must not be negative", e.getMessage());
+        }
```

Comment:
Itt azt teszteled le, hogy HA dob kivetelt, aznnak ez legyen az uzenete. Azt nem, hogy dob-e kivetelt. Ami ki van kommentelve, az jo lenne, ha mukodesre lenne birva.

Reply:
Ide én is hozzátenném, hogy általában kivételek teszteléséhez az AAA kb így néz ki:
```java
// Arrange
Executable executable;

// Act
executable = () -> new Entity(...);

// Assert
assertThrows(...);
```
`Executable` interfész elérhető a `org.junit.jupiter.api.function` csomagból.

Reply:
@hegyhati 

Megcsináltam amiket kértél.
- az attack-ot teszteltem, előszőr a saját_támadás - másik_védelme logika alapján, és megnéztem, hogy az elvárt HP-t kapjuk e.
- másodjára már az attack()-al és itt is megnéztem, hogy a HP-k a kívántra változnak e.
- ">=" átírtam arra, amit várunk.
- egyszerűsítettem a booleanosokat assertTrue és assertFalse-ra. 
- az exception-t is megcsináltam, és most jó (ami ki volt kommentelve ott nem jöttem rá, mi volt a gond, mert X-et kapott tesztelésnél, ezért változtattam meg try-catch-re, ezért elnézést kérek).